pkg/odo/cli/dev/dev.go

```diff
@@ -193,29 +234,64 @@ func (o *DevHandler) RegenerateAdapterAndPush(pushParams common.PushParameters,
 }
 
 func regenerateComponentAdapterFromWatchParams(parameters watch.WatchParameters) (common.ComponentAdapter, error) {
-
-	// Parse devfile and validate. Path is hard coded because odo expects devfile.yaml to be present in the pwd/cwd.
-
 	devObj, err := ododevfile.ParseAndValidateFromFile(location.DevfileLocation(""))
 	if err != nil {
 		return nil, err
 	}
 
+	changed, err := haveEndpointsChanged(parameters.InitialDevfileObj, devObj)
+	if err != nil {
+		return nil, err
+	} else if changed {
```

Comment:
you don't need this `else`

Reply:
I think we do because if there is no change in the endpoints, we don't want to print the subsequent message to do `odo dev` again.

Reply:
I mean you can write `if changed` instead of `else if changed`, as you returned in the `if`